File: C_Programming/Practise_Programs/PP_156.c
//////////////////////////////////////////////////////////////////
//
//  File Name         : PP_156
//
//  Description       : This program demonstrates the left shift
//                      operation on an unsigned integer and displays
//                      the result in hexadecimal format.
//
//  Author            : Shubham Somanath Gadhe
//  Date              : 19/08/2026
//
//  Time Complexity   : O(1)
//  Space Complexity  : O(1)
//
//////////////////////////////////////////////////////////////////

#include<stdio.h>

typedef unsigned int UNIT;

int main()
{
    UNIT iMask = 0x1;
    int iCnt = 0;

    printf("%X\n", iMask);  
    iMask = iMask << 1;

    printf("%X\n", iMask);  
    iMask = iMask << 1;

    printf("%X\n", iMask);  
    iMask = iMask << 1;

    printf("%X\n", iMask);  
    iMask = iMask << 1;

    printf("%X\n", iMask);  
    iMask = iMask << 1;

    printf("%X\n", iMask);  
    iMask = iMask << 1;

    printf("%X\n", iMask);  
    iMask = iMask << 1;

    printf("%X\n", iMask);  
    iMask = iMask << 1;

    printf("%X\n", iMask);  
    iMask = iMask << 1;

    printf("%X\n", iMask);  
    iMask = iMask << 1;

    printf("%X\n", iMask);  
    iMask = iMask << 1;

    printf("%X\n", iMask);  
    iMask = iMask << 1;

    printf("%X\n", iMask);  
    iMask = iMask << 1;

    return 0;
}

//////////////////////////////////////////////////////////////////
//
//  Sample Input :
//      No input required.
//
//  Sample Output :
//
//      1
//      2
//      4
//      8
//      10
//      20
//      40
//      80
//      100
//      200
//      400
//      800
//      1000
//
//////////////////////////////////////////////////////////////////

File: Java_Programming/Practise_Programs/PP_156.java
//////////////////////////////////////////////////////////////////
//
//  File Name         : PP_156
//
//  Description       : This program accepts a number from the
//                      user and displays the number for odd
//                      positions and a star for even positions.
//
//  Author            : Shubham Somanath Gadhe
//  Date              : 03/09/2026
//
//  Time Complexity   : O(n)
//  Space Complexity  : O(1)
//
//////////////////////////////////////////////////////////////////

import java.util.*;

class PP_156
{
    public static void Display(int iNo)
    {
        int iCnt = 0;

        for(iCnt = 1; iCnt <= iNo; iCnt++)
        {
            if(iCnt % 2 == 0)
            {
                System.out.print("*\t");
            }
            else
            {
                System.out.print(iCnt + "\t");
            }
        }
        System.out.println();
    }
    public static void main(String A[])
    {
        Scanner sobj = new Scanner(System.in);

        int iValue = 0;

        System.out.println("Enter the elements : ");
        iValue = sobj.nextInt();

        Display(iValue);        
    }
}

//////////////////////////////////////////////////////////////////
//
//  Sample Input :
//      Enter the elements : 5
//
//  Sample Output :
//
//      1	*	3	*	5
//
//////////////////////////////////////////////////////////////////